src/account.rs

```diff
@@ -67,6 +72,41 @@ where
             Self::Sub { .. } => todo!(),
         }
     }
+
+    /// Save data to be signed later
+    pub fn add_to_pending_sign(&mut self, message: &[u8]) -> Result<()> {
+        match self {
+            Self::Root { pending_sign, .. } | Self::Sub { pending_sign, .. } => 
+                pending_sign.borrow_mut().push_back(message.into()),
+        };
+        Ok(())
+    }
+
+    /// Try to sign messages from the queue
+    /// Return signed messages
+    pub fn sign_all_pending(&mut self) -> Vec<(Vec<u8>, P::Signature)> {
+        let mut signed = Vec::new();
+        let pending = match self {
+            Self::Root { pending_sign, .. } | Self::Sub { pending_sign, .. } => pending_sign.clone()
+        };
+        while !pending.borrow().is_empty() {
```

Comment:
As I mentioned in above, there is `drain`(for both VecDeque and Vec) that we can use to empty the collection. A more functional alternative that uses drain can look something like:
```rust
pending.drain(..).map(|msg| (msg, self.sign(&msg))).collect()
```
We can even avoid allocating the Vec ourselves if we omit the `collect` and return the iterator instead(i.e. `-> impl Iterator<Item=(Vec<u8>, P::Signature)>`) :)